crates/eth_compatibility/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_std]
+use ink_env::{
+    DefaultEnvironment,
+    Environment,
+};
+
+/// The ECDSA compressed public key.
+#[derive(Debug, Copy, Clone)]
+pub struct ECDSAPublicKey(pub [u8; 33]);
+
+impl Default for ECDSAPublicKey {
+    fn default() -> Self {
+        // Default is not implemented for [u8; 33], so we can't derive it for ECDSAPublicKey
+        // But clippy thinks that it is possible. So it is workaround for clippy.
+        let empty = [0; 33];
+        Self { 0: empty }
```

Comment:
If it looks like:
```rust
impl Default for ECDSAPublicKey {
    fn default() -> Self {
        Self { 0: [0; 33] }
    }
}
```

Clippy in CI will fail. Because it says that I can derive `Default` because the output of deriving and my manual implementation are the same. But when I'm deriving `Default`, it fails because `Default` is not implemented for `[u8; 33]`. So it is a workaround to pass your CI=)